addons/l10n_sa_edi/models/account_edi_xml_ubl_21_zatca.py

```diff
@@ -371,10 +371,10 @@ def _get_invoice_line_vals(self, line, line_id, taxes_vals):
 
         def grouping_key_generator(base_line, tax_values):
             tax = tax_values['tax_repartition_line'].tax_id
-            tax_category_vals = self._get_tax_category_list(line.move_id, tax)[0]
+            tax_category_vals = next(iter(self._get_tax_category_list(line.move_id, tax)), {})
```

Comment:
I think the tax category is mandatory in ubl... (to test). If it's the case, it's probably better to raise or to change the code of `_get_tax_category_list` to find at least one category for that case

Reply:
I agree with @smetl :+1:
In all formats that I've seen in UBL, this node is mandatory.